### DISCUSSION TITLE
(GitHub Enterprise Cloud) Search across all organizations in an enterprise

### DISCUSSION THREAD

**Discussion by raja-sekar-db** [community/community#49510] [Enterprise] [answered]
### Select Topic Area

Question

### Body

I work for a large enterprise & a GitHub Enterprise user. There are 100s of GitHub Organizations in this enterprise setup. 

The GitHub Code Search capability only allows the following scopes
- within a repo
- within an organization
- all GitHub 

Is it possible to scope the search to all organizations (only) in my enterprise?

**Comment by JoshuaTheMiller** [comment]
As of now and as far as I'm aware, **no- it is not possible** 😟 With that being said, hopefully more folks come across this Discussion and Vote on it so that GitHub knows that it is a highly useful feature (one that is _probably_ highly requested as well!).

In case folks say otherwise- yes, we could create a massive search-string that includes a list of all Enterprise Orgs as Org filters. With large Enterprises though, this gets unwieldy and we **will** bump into search size limits.

**Reply by satyakrish** [reply]
Upvoted, we definitely need this ability to search only within our enterprise.

**Comment by justinmchase** [comment]
If anyone from GitHub reads this I would propose a new search term `ent:<name>`  where `<name>` is the name of the enterprise you wish to search in.

Also, sorry but I'm the one filing the feedback via the feedback button about this over and over again, as I keep getting burned by it :)


![Screen Shot 2023-04-03 at 9 19 28 AM](https://user-images.githubusercontent.com/10974/229538379-0a39c147-9a93-4095-b08c-31c949c795ed.png)

**Comment by look** [comment]
Thanks for your patience on this. We hear you and plan to implement it, but we've been focusing on getting the new search scaled up so everyone can use it.

**Reply by justinmchase** [reply]
> ...new search...

Is there anywhere where we can read more about this?

**Reply by look** [reply]
Yes, you can check out the [feature page](https://github.com/features/code-search) and our [announcement blog post](https://github.blog/2023-05-08-github-code-search-is-generally-available/).

**Reply by Timmmy-nlb** [reply]
After 10 months I would like to ask: what about this feature?
Unfortunately I can't find anything on the github roadmap... has this feature request already been canceled?

**Reply by justinmchase** [reply]
Yes please, @look any update on the enterprise search specifically?

**Reply by look** [reply]
We've had to prioritize other work, but we still plan to implement this.

**Reply by nearlythere** [reply]
This question looks "answered" but it's not really answered is it? Is there a roadmap item we can subscribe to?

It would be really handy to be able to investigate incidents with an enterprise-wide search tool. Otherwise, we run into rate limits, or other limitations.

**Reply by Timmmy-nlb** [reply]
> It would be really handy to be able to investigate incidents with an enterprise-wide search tool. Otherwise, we run into rate limits, or other limitations.

And this limit is VERY low!

**Reply by brendan-kellam** [reply]
(_shameless plug_) If you are looking for an alternative solution, I've built a standalone code search tool that has already solved this issue for other GHEC users 🤠  

https://github.com/sourcebot-dev/sourcebot

**Reply by lvthillo** [reply]
@look I understand you probably can't give an ETA, but is this still visible on your roadmap? I don't see any reference [here](https://github.com/orgs/github/projects/4247).

**Comment by rdohms** [comment]
@look has there been any progress on getting this into the backlog? Our Org is really hindered by it, and we are already going over the char limit for adding `org:` filters as a workaround.

**Reply by look** [reply]
We've heard the feedback that this is an important feature for many customers, and we have added it to our near-term roadmap. Sorry I can't be more precise, but it is coming!

**Reply by rdohms** [reply]
How does it look @look? How far in the near-term is it?

**Reply by rdohms** [reply]
guess is not looking so good...

**Comment by emilva** [comment]
+1. We will not migrate to GHEC before this feature is available.

**Reply by emilva** [reply]
> +1. We will not migrate to GHEC before this feature is available.

+1 upvoting myself again 2 years later. Please prioritize this feature.

**Comment by jennifer-blate-db** [comment]
The new search is waaay better than the old search, but it still doesn't have this functionality, which would be very helpful.

**Comment by jennifer-blate-db** [comment]
Is there a more official location that we can upvote this feature request or is this it?

**Reply by look** [reply]
This is the best place. I hope to be able to share updates soon. 😉

**Reply by jennifer-blate-db** [reply]
Great!  looking forward to it!

**Reply by brettshearer** [reply]
Any updates yet?

**Comment by MartinLesterSynamedia** [comment]
You can get this data using the GraphQL API

```
query get_orgs($enterprise:String!, $pos:String) {
    enterprise(slug: $enterprise) {
        organizations(first: 100, after: $pos) {
           edges{
               node {
                  login
               }
            }
            pageInfo {
                hasNextPage
                endCursor
            }
        }  
    }
}
```

```
{
  "enterprise": "ENTERPRISE-NAME",
  "pos": ""
}
```
Try it with the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer)

When using pagination, you need to keep calling the query with `pos = endCursor` till `hasNextPage` is false.

**Reply by eisenhowerj** [reply]
I think you misunderstood the request. We don't need a list of all organizations within an enterprise account; we need the ability to _search_ across all orgs e.g. this https://docs.github.com/en/graphql/reference/queries#search

**Comment by emilva** [comment]
FYI: the search in the GHEC EU data residency version is actually scoped by default to your enterprise, and there are no github.com results mixed in. https://github.com/newsroom/press-releases/data-residency-in-the-eu

**Reply by vegardx** [reply]
Sort of, but it only partially solves the issue. There's no way to do enterprise-wide searches from the search within an organization, this can only be done from the enterprise landing page or a separate search page within your tenant, even though the UX for the search is identical, the functionality is widely different. And from a users perspective this makes no sense. How you'd effectively communicate this to your enterprise users I struggle to see. 

My guess would be that GitHub does something similar to how GitHub Enterprise Cloud with Data Residency for GitHub Enterprise Cloud in general. I never understood why you'd even want to have GitHub Enterprise Managed Users on github.com, it just confuses people, gives you headaches with people not being able to use the same SSH key and whatnot.

**Reply by justinmchase** [reply]
> ...starting with general availability in the European Union (EU) on October 29, 2024. GitHub also confirms that additional regions are coming next, including Australia, Asia, and Latin America. 

![yudodis](https://github.com/user-attachments/assets/b00c95b7-787e-4b63-b8c7-75f18ba26bb1)

USA when?

**Comment by gitdmb** [comment]
Adding an upvote for this FR, Yahoo! continues to make this request. Is there an existing feature request issue I should be following? I couldn't find one

**Comment by kakwa** [comment]
Could you had a roadmap item for this feature?

I'm not finding it in:

https://github.com/orgs/github/projects/4247/views/1?filterQuery=search

**Comment by Urkasrk** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> I work for a large enterprise & a GitHub Enterprise user. There are 100s of GitHub Organizations in this enterprise setup.
> 
> The GitHub Code Search capability only allows the following scopes
> 
> * within a repo
> * within an organization
> * all GitHub
> 
> Is it possible to scope the search to all organizations (only) in my enterprise?

**Reply by Bashirsaba** [reply]
Ok sir

**Comment by queenofcorgis** [comment, marked as answer]
Thank you everyone for your patience and the feedback which helped us ship this feature, you can now search across Enterprises, we [updated our docs to include this](https://docs.github.com/en/enterprise-cloud@latest/search-github/github-code-search/understanding-github-code-search-syntax#enterprise-qualifier), stay tuned for the changelog!

**Reply by WyriHaximus** [reply]
@queenofcorgis Awesome, really really happy to see this feature land :heart: ! One note tho, if your enterprise has an understore in it say `my_enterprise` you need to use `enterprise:myenterprise` instead of `enterprise:my_enterprise` to make it work.

**Reply by jorendorff** [reply]
Thanks for pointing this out.

For anyone having trouble figuring out what enterprise ID to use: Go to any repository in your enterprise and click in the search bar or press `/`. You'll see some search suggestions, like this:

<img width="635" height="224" alt="search suggestions" src="https://github.com/user-attachments/assets/24f56e8f-ae2b-4374-9e2f-5bef9c54d184" />

One of them will show the `enterprise:` syntax to use for your enterprise.

**Reply by WyriHaximus** [reply]
Correct, that is how we found and realised this. The other thing that caught us off guard a bit is that this is only for code search, looking at this topic that makes sense 😅 .

**Reply by gjasny** [reply]
Thanks for implementing that. Could you please give an example how to use the `enterprise:myenterprise` in  a `gh search code` or `gh api` command call?

**Reply by look** [reply]
Unfortunately, this is not available in the API at this time, so using `gh` will not work.

**Comment by pedramalizade** [comment]
You can’t scope GitHub Code Search to “all organizations in your Enterprise only.”
Currently, the available scopes are:

repo: specific repositories

org: a single organization

global: everything your account has access to (all orgs + all repos)

GitHub Enterprise doesn’t provide an “enterprise-level only” search filter.
If you want to limit results to your Enterprise organizations, the closest workaround is:

Run a global search

Add org:ORG-NAME filters for the orgs you specifically want

(Optional) Create a custom list of orgs and reuse that search query

It’s not perfect, but right now GitHub doesn’t expose a direct “enterprise scope” for Code Search.

If GitHub adds an Enterprise-wide search scope in the future, it should appear in the official Code Search docs.

**Reply by look** [reply]
We have shipped the ability to limit your search to an enterprise with the `enterprise:` qualifier ([docs](https://docs.github.com/en/enterprise-cloud@latest/search-github/github-code-search/understanding-github-code-search-syntax#enterprise-qualifier)).

**Reply by pedramalizade** [reply]
Thanks for the update! Glad to see the enterprise: qualifier is now available—it will make Enterprise-wide searches much easier.

**Comment by CodeWithMehru** [comment]
**You can now search across all organizations in an enterprise using the new enterprise qualifier in GitHub Code Search. Use it like this for example

<search term> enterprise:your-enterprise-name


This will return results from all organizations that belong to that enterprise and that you have access to. The qualifier is generally available to GitHub Enterprise Cloud customers and behaves like other search qualifiers so it keeps the same limits and permission rules. 
If you do not have the qualifier yet the workaround is to run a global search and add multiple org: qualifiers or use an external code search tool like Sourcegraph that can index many organizations.**

**Comment by nejoyer** [comment]
I love that I can search for code across the enterprise. This is invaluable!

I'd love to be able to search for repos the same way, but it looks like it isn't supported. I hope this is something that can be added soon!

<img width="424" height="44" alt="image" src="https://github.com/user-attachments/assets/5e6a8ede-8cba-4c0b-854d-3ba540ffff32" />

**Reply by dmitris** [reply]
> I'd love to be able to search for repos the same way, but it looks like it isn't supported. I hope this is something that can be added soon!

I'd love / really need to be able to search for Pull Request across the enterprise.
<img width="768" height="106" alt="Screenshot 2026-01-20 at 19 44 41" src="https://github.com/user-attachments/assets/f31b95e8-1daf-4449-8249-921785211dd9" />

**Comment by mwisnicki** [comment]
`enterprise:` seems to only work for code search, all other categories always show 0 results.

**Reply by mwisnicki** [reply]
Ok, I see, when I go to any category it actually shows a warning:
<img width="714" height="77" alt="image" src="https://github.com/user-attachments/assets/1fae1c00-abcc-4557-aee9-17363a38a53c" />

That's disappointing.
